lib/beaker/hypervisor/vagrant.rb

```diff
@@ -23,10 +23,13 @@ def private_network_generator(host)
       private_network_string = "    v.vm.network :private_network, ip: \"#{host['ip'].to_s}\", :netmask => \"#{host['netmask'] ||= "255.255.0.0"}\""
       case host['network_mac']
       when 'false'
+        @mac = nil
```

Comment:
@bastelfreak why don't you set `@mac` to be `randmac` here as well? Since you've kept the `private_network_string` appending in the case statement, it doesn't seem necessary to set this to `nil` now & then have the ternary statement (on line 108 below). Is there a use of `@mac` between here & there, or something after that would be poisoned that I'm missing?

You might even be able to pull both `false` & `nil` cases where you set `@mac = randmac` out of the case statement in general until just after it with something like `@mac ||= randmac`.

Also, I'm noticing that in the two cases below this one, you're assigning values to `@mac` & then using them in the `private_network_string` append. It would be better to use the `@mac` value rather than the right side of the assignments statement, as in the future if there's a reason to change what `@mac` is set to, we'll be able to do it in one place for each of these cases rather than in two.

I'm sure you'll notice the last two pieces of feedback are contrary to each other, as you'd need to have the `@mac` assignments above the `private_network_string` append in order to use the `@mac` values in the append lines. I don't expect you to do both 😄 . Which seems cleanest to you?

Reply:
I don't see how I should be able to pull the assignments out of the case statement.  I need to write the generated mac address into private_network_string in line 28. I wouldn't really gain much from pulling them out?